Sources/TuistGenerator/Utils/SettingsHelper.swift

```diff
@@ -44,4 +38,34 @@ final class SettingsHelper {
     func variant(_ buildConfiguration: BuildConfiguration) -> BuildSettingsProvider.Variant {
         return buildConfiguration.variant == .debug ? .debug : .release
     }
+
+    // MARK: - Private
+
+    private func resolveValue(oldValue: SettingValue?, newValue: SettingValue) -> SettingValue {
+        guard let oldValue = oldValue else {
+            return newValue
+        }
+        guard oldValue != newValue else {
+            return oldValue
+        }
+
+        switch (oldValue, newValue) {
+        case let (.string(old), .string(new)) where new.contains("$(inherited)"):
```

Comment:
I'd add a comment on top of each case with an example of which use-case each line is dealing with. As @ollieatkinson pointed out, this is not easy to follow.

Reply:
Added some comments and examples. Hopefully it makes the logic easier to follow.